src/components/PlantingSites/editor/Zones.tsx

```diff
@@ -77,9 +106,39 @@ export default function Zones(props: ZonesProps): JSX.Element {
 
       const cutWith = featureCollection?.features?.[0]?.geometry;
       if (cutWith) {
-        const newZones = cutPolygons(zones.features as GeometryFeature[], cutWith);
-        if (newZones) {
-          setZones({ type: 'FeatureCollection', features: newZones });
+        const cutZones = cutPolygons(zones.features as GeometryFeature[], cutWith);
+
+        if (cutZones) {
+          const idGenerator = IdGenerator(cutZones);
+          const zonesWithIds = cutZones.map((zone) => toZoneFeature(zone, idGenerator)) as GeometryFeature[];
+
+          setZones({
+            type: 'FeatureCollection',
+            features: zonesWithIds,
+          });
+
+          const leftMostNewZone = zonesWithIds
```

Comment:
This seems like a good candidate for its own util function and a test

Reply:
Good call, will update this PR with util/test and also the cutPolygons test.